### ISSUE TITLE
Delete-until-character operations incorrectly move cursor to new line

### ISSUE THREAD

**Issue by BlackLight** [vscode-neovim/vscode-neovim#818]
Example source code:

```javascript
if (foo == 'bar')
  console.log('ok')
```

- Place the cursor on the letter `b`
- Perform any vi edit-until-character operations (e.g. `dt'` or `df'`)

**Expected**

The cursor should remain on the letter `b` after the deletion is performed (or, at least, this is how vi/vim/neovim behave).

**Actual**

The cursor, for some reason, is moved to the beginning of the next line.

**Note**

Only the `d` command with cursor motion instructions (i.e. `dt*` and `df*`) seem to be affected.

`di*`, `ci*`, `vt*` etc. are not affected by this behaviour (the cursor doesn't jump to the next line after execution).

**Comment by asvetliakov** [comment]
not reproducible. try with empty init.vim

**Comment by BlackLight** [comment]
The issue seems to have been resolved by itself somehow. I tried with an empty `init.vim` and everything worked, then I tried to restore my `init.vim` (the exact same one that replicated the issue) and everything still worked. That's a bit weird - I'll close this issue for now and I'll reopen it if the issue pops out again.

**Comment by BlackLight** [comment]
@asvetliakov the issue had previously gone only because I had commented large swaths of configuration in my `init.vim`.

Once I uncommented everything and restored all the plugins, the issue came back.

After a lot of bisecting to find the culprit, I realized that the issue isn't caused by any of my custom mappings but by a vim plugin - in my case [yankring](https://github.com/vim-scripts/YankRing.vim).

When the plugin is disabled, typing e.g. `dt.` works as expected. When the plugin is enabled, the command moves the cursor to the line below.

Since I haven't really used that plugin in ages (and the last commit on that repo is also >7 years old), removing it isn't a big deal for me, and I haven't really investigated what's happening under the hood that messes the state of the cursor only in vscode but not in nvim.
